inc/dashboard/namespace.php

```diff
@@ -0,0 +1,456 @@
+	}
+
+	return $xb_columns;
+}
+
+/**
+ * Update the Block column's orderby value.
+ *
+ * This displays the title, but the column is a custom column called "block", so we need to change the orderby value to "title".
+ *
+ * @param array $vars An array of query variables passed in from the filter.
+ *
+ * @return array The updated query variables.
+ */
+function xb_block_column_orderby( $vars ) : array {
+	if ( isset( $vars['orderby'] ) && 'block' === $vars['orderby'] ) {
+		$vars['orderby'] = 'title';
+	}
+
+	return $vars;
+}
+
+/**
+ * Render links for date range data.
+ */
+function render_date_range_links() {
+	$ranges = [
+		[
+			'days' => 7,
+			'label' => __( '7 days', 'altis-analytics' ),
+		],
+		[
+			'days' => 30,
+			'label' => __( '30 days', 'altis-analytics' ),
+		],
+		[
+			'days' => 90,
+			'label' => __( '90 days', 'altis-analytics' ),
+		],
+	];
+
+	$days = get_days_view();
+	?>
+	<div class="xb-analytics-date-range">
+		<?php
+		foreach ( $ranges as $current ) {
+			// Compare the current days value with the current days value pulled from the url query variables.
+			if ( $days === $current['days'] ) {
+				$selected = 'selected';
+			} else {
+				$selected = '';
+			}
+
+			printf(
+				wp_kses_post( '<a href="%1$s" class="date-range-button %2$s">%3$s</a>' ),
+				esc_url_raw( add_query_arg( [ 'days' => $current['days'] ] ) ),
+				esc_html( $selected ),
+				esc_html( $current['label'] )
+			);
+		}
+		?>
+	</div>
+	<?php
+}
+
+/**
+ * Render the block name column data.
+ */
+function render_block_column() {
+	global $post;
+	?>
+	<strong><a href="<?php echo esc_url_raw( admin_url( '/admin.php?page=xb-analytics&post=' . $post->ID ) ); ?>"><?php echo esc_attr( $post->post_title ); ?></a></strong>
+	<?php
+}
+
+/**
+ * Render the views column data.
+ */
+function render_views() {
+	global $post;
+	$list = get_views_list( get_days_view() );
+	$views = $list[ $post->post_name ]['views'];
+	?>
+	<div class="post--views"><?php echo number_format_i18n( $views ); ?></div>
+	<?php
+}
+
+/**
+ * Render the average conversion rate column data.
+ */
+function render_average_conversion_rate() {
+	global $post;
+	$list = get_views_list( get_days_view() );
+	$rate = round( $list[ $post->post_name ]['avg_conversion_rate'] * 100 );
+	?>
+	<div class="post--avg-conversion-rate"><?php echo absint( $rate ); ?>%
+	<?php
+}
+
+/**
+ * Return the days url query string.
+ *
+ * @return int The number of days to show insights for. Defaults to 7.
+ */
+function get_days_view() : int {
+	return isset( $_GET['days'] ) ? wp_unslash( absint( $_GET['days'] ) ) : 7;
+}
+
+/**
+ * Calculate average conversion rate.
+ *
+ * @param int $conversions The number of conversions for a block.
+ * @param int $views The number of views for a block.
+ *
+ * @return float The conversion rate (calculated by conversions / views).
+ */
+function calculate_average_conversion_rate( int $conversions = 0, int $views = 0 ) : float {
```

Comment:
Can remove this function now.

Reply:
I was hanging onto it until I knew for sure the ES query for that was working and then forgot to take it out. 😅